src/database/DataTypes/StocktakeBatch.js

```diff
@@ -211,7 +211,25 @@ export class StocktakeBatch extends Realm.Object {
    */
   set countedTotalQuantity(quantity) {
     // Handle packsize of 0.
-    this.countedNumberOfPacks = this.packSize ? quantity / this.packSize : 0;
+    const countedNumberOfPacks = this.packSize ? quantity / this.packSize : 0;
+
+    this.countedNumberOfPacks = countedNumberOfPacks;
+    this.doses = countedNumberOfPacks * this.dosesPerVial;
+  }
+
+  get isVaccine() {
+    return this.itemBatch?.item?.isVaccine ?? false;
+  }
+
+  get dosesPerVial() {
+    return this.isVaccine ? this.itemBatch?.item?.doses ?? 0 : 0;
+  }
+
+  setDoses(database, newValue) {
+    const maximumDosesPossible = this.dosesPerVial * this.countedTotalQuantity;
+    this.doses = Math.min(newValue, maximumDosesPossible);
+
+    database.save('StocktakeBatch', this);
```

Comment:
setting doses: just have a maximum rather than a minimum, otherwise entering details gets messy. I.e. if you enter quantity = 10 with number of doses = 2, the number of doses given should really be between 10 and 20. But you can't enter `1`.. so just enforce the max. Not sure if it should check on finalising. Probably should, I'll make an issue

Reply:
Don't know much about this logic, but if `dosesPerVial` can be any significant number, couldn't this lead to stocktakes indicating a lot more doses are available than actually are? Could that cause inventory issues down the line?

Reply:
there are a few comments here regarding doses etc: 

- `item.doses` is the number of doses in a pack of an item. 
- Example item: `VaccineA`
- `item.doses` is the number of doses in a vial of `VaccineA` [Lost the fight on calling it `dosesPerVial` :( ..! ]. So if `item.doses == 2`, each vial of `VaccineA` has two doses in it. 
- Vials in general have a very short life span after being opened, so there is often open vial wastage, resulting in vials being only 'half used'
- `..Batch.doses` is tracking the actual number of doses. 
- The `ItemBatch`s for `VaccineA` have `doses` which is the number of doses currently available - so it should theoretically  always be `item.doses * numberOfPacks * packSize`, but with the wastage from a vial, so it could be less. 
- `TransactionBatch` have a `doses` field which is tracking how many doses were in some transaction.
- `StocktakeBatch` have a `doses` field which is tracking how many doses of a batch are left. There are vaccines which have 10 doses. If you gotta chuck it out with 9 doses left in it, you can make an inventory adjustment in the stocktake which reduces the number of doses by 9, if you want, rather than a full pack.

probably more confusing now, but this sets an upper limit on the `StocktakeBatch` for the number of doses. You're right though - the number of doses won't really ever be that accurate!